modules/imgcodecs/src/grfmt_base.cpp

```diff
@@ -56,6 +56,8 @@ BaseImageDecoder::BaseImageDecoder()
     m_scale_denom = 1;
     m_use_rgb = false;
     m_frame_count = 1;
+    m_metadata_reading_flag = 0;
+    m_metadata.resize(IMAGE_METADATA_MAX + 1);
```

Comment:
Why do you need +1 here?

Reply:
```
enum ImageMetadataType
{
    IMAGE_METADATA_UNKNOWN = -1,  // Unrecognized
    IMAGE_METADATA_EXIF    = 0,
    IMAGE_METADATA_XMP     = 1,
    IMAGE_METADATA_ICCP    = 2,
    IMAGE_METADATA_TEXT    = 3,

    IMAGE_METADATA_MAX     = 3    // Highest legal index (NOT a count)
};
```

// Allocate metadata storage
std::vector<std::vector<uchar>> m_metadata(IMAGE_METADATA_MAX + 1);

// Access
m_metadata[IMAGE_METADATA_EXIF];  // 0
m_metadata[IMAGE_METADATA_XMP];   // 1
m_metadata[IMAGE_METADATA_ICCP];  // 2
m_metadata[IMAGE_METADATA_TEXT];  // 3

Reply:
Got it, thanks!